webapp/channels/src/utils/cloud_utils.ts

```diff
@@ -47,9 +47,22 @@ export function openExternalPricingLink() {
 
 export const FREEMIUM_TO_ENTERPRISE_TRIAL_LENGTH_DAYS = 30;
 
-export function daysToExpiration(expirationDate: number): number {
-    const now = new Date();
-    const expiration = new Date(expirationDate);
+export function daysUntil(end?: number, simulatedCurrentTimeMs?: number) {
+    let now = new Date();
+
+    if (simulatedCurrentTimeMs) {
+        now = new Date(simulatedCurrentTimeMs);
+    }
+
+    const expiration = new Date(end || 0);
     const diff = expiration.getTime() - now.getTime();
     return Math.ceil(diff / (1000 * 3600 * 24));
 }
+
+export function daysToExpiration(subscription?: Subscription): number {
```

Comment:
Does `subscription` need to be optional here? At least where I see it used in `billing_subscriptions.tsx`, it's null-checked before usage. If it's passed in as undefined elsewhere, this is fine as-is, but I think it's a good idea to try to make as few things as possible optional since it avoids spreading question marks all over the code.

The same sort of thing would apply above to these other functions too

Reply:
It's mainly here that it's possibly undefined: https://github.com/mattermost/mattermost/pull/25927/files#diff-96273a2b9a83141ddf1b497fc686aff77ce153a80b27e0aadbf0196cd30d2e5cR37